app/src/main/java/de/danoeh/antennapod/dialog/FilterDialog.java

```diff
@@ -21,42 +29,143 @@ public FilterDialog(Context context, FeedItemFilter feedItemFilter) {
         this.filter = feedItemFilter;
     }
 
-    public void openDialog() {
-        final String[] items = context.getResources().getStringArray(R.array.episode_filter_options);
-        final String[] values = context.getResources().getStringArray(R.array.episode_filter_values);
-        final boolean[] checkedItems = new boolean[items.length];
+    public void openCustomDialog() {
 
         final Set<String> filterValues = new HashSet<>(Arrays.asList(filter.getValues()));
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(R.string.filter);
 
-        // make sure we have no empty strings in the filter list
-        for (String filterValue : filterValues) {
-            if (TextUtils.isEmpty(filterValue)) {
-                filterValues.remove(filterValue);
-            }
-        }
+        LayoutInflater inflater = LayoutInflater.from(this.context);
+        View customLayout = inflater.inflate(R.layout.custom_filter_dialog_layout, null);
+        TableLayout tableLayout = customLayout.findViewById(R.id.custom_dialog_tableLayout1);
+
+        builder.setView(customLayout);
+
+        for (FeedItemEnum item : FeedItemEnum.values()) {
+
+            RelativeLayout row = (RelativeLayout) inflater.inflate(R.layout.custom_dialog_filter_row, null);
+            TextView textView1 = row.findViewById(R.id.custom_dialog_textView1);
+            TextView textView2 = row.findViewById(R.id.custom_dialog_textView2);
+            TextView textView3 = row.findViewById(R.id.custom_dialog_textView3);
+            textView1.setText(item.values[1].displayName);
+            textView1.setTextColor(Color.DKGRAY);
+            textView2.setText(item.values[0].displayName);
+            textView2.setTextColor(Color.DKGRAY);
 
-        for (int i = 0; i < values.length; i++) {
-            String value = values[i];
-            if (filterValues.contains(value)) {
-                checkedItems[i] = true;
+            Iterator<String> filterIterator = filterValues.iterator();
+            while (filterIterator.hasNext()) {
+                String nextItem = filterIterator.next();
+                if (item.values[1].filterId.equals(nextItem)) {
+                    item.values[1].setSelected(true);
+                    item.values[0].setSelected(false);
+                    textView1.setSelected(true);
+                    textView2.setSelected(false);
```

Comment:
If you use `RadioButtons` instead of `TextViews`, you do not need to un-check other items in the row manually. They can be styled just like the TextViews (background etc). Details: https://stackoverflow.com/a/55139080/4193263

Reply:
radio buttons it is